src/swarm/neo/connection/RequestOnConnBase.d

```diff
@@ -507,6 +614,183 @@ abstract class RequestOnConnBase
 
         /***********************************************************************
 
+            Waits for one of the specified events to occur and returns a
+            smart-union denoting which occurred.
+
+            There are four possible types of event:
+                1. Receiving a payload for this request.
+                2. Sending a payload.
+                3. The fiber being resumed after it was yielded and an epoll
+                   event-loop cycle occurred.
+                4. The fiber being resumed with a non-negative code.
+
+            The fiber is usually suspended while waiting for an event to occur.
+            The one exception is when sending -- the send may succeed
+            immediately, without needing to suspend the fiber.
+
+            Params:
+                flags = flags indicating whether to wait for receiving &/
+                    being resumed after yielding
+                fill_payload = optional delegate to fill in a payload to send.
+                    If this argument is null, sending does not occur
+
+            Returns:
+                an EventNotification instance denoting the event which occurred
+
+            Throws:
+                Exception on protocol or I/O error.
+
+        ***********************************************************************/
+
+        public EventNotification nextEvent (
```

Comment:
Yeah, better try it. This "can't have two contexts" thing is floating around in my head. If worst comes to worst, we can always pass just that one as a runtime arg.